lib/logstash/filters/http.rb

```diff
@@ -152,4 +153,15 @@ def process_response(body, headers, event)
     end
   end
 
+  ##
+  # @param headers [String] or [Array]
+  # @return resolved content-type
+  def headers_has_json_content_type?(headers)
+    # content-type might be an array or string with ; separated
+    [ headers.fetch("content-type", "") ]
+             .map {|i| i }
+             .flatten
+             .include?("application/json")
+  end
+
```

Comment:
```suggestion
  ##
  # @param headers [String] or [Array]
  # @return boolean
  def headers_has_json_content_type?(headers)
    # content-type might be an array or string with ; separated
    headers = headers.kind_of?(Array) ? headers : headers.split(';')
    return headers.map(&:strip).include?("application/json")
  end

```

Reply:
I have updated the suggestion with working (locally tested) version.